lib/rspec/rails/fixture_file_upload_support.rb

```diff
@@ -0,0 +1,41 @@
+module RSpec
+  module Rails
+    # @private
+    module FixtureFileUploadSupport
+      extend ActiveSupport::Concern
+
+      included do
+        delegate :fixture_file_upload, :to => :rails_test_wrapper
+      end
+
+    private
+
+      def rails_test_wrapper
+        resolved_fixture_path = (fixture_path || RSpec.configuration.fixture_path || '')
+        RailsTestWrapper.fixture_path = File.join(resolved_fixture_path, '')
+        RailsTestWrapper.instance
+      end
+
+      class RailsTestWrapper
+        class << self
+          attr_reader :fixture_path
+
+          def instance
+            @instance ||= new
+          end
+
+          def fixture_path=(value)
+            if ActionController::TestCase.respond_to?(:fixture_path)
+              ActionController::TestCase.fixture_path = value
+            end
+            @fixture_path = value
```

Comment:
Do we need to store the value if we can't set it? I think it'd be cleaner if we just implemented this in the main mix in and delegated directly to this with the resolved file.

Reply:
yes because `ActionController::TestCase.fixture_path` is expecting class to respond to `fixture_path`
And I have to override setter method because of Rails 3 implementation of `fixture_file_upload`.

Reply:
@JonRowe CI seems to have some random failures / between 2 builds with exact same code differents environments are failing

Reply:
> yes because ActionDispatch::TestProcess is expecting class to respond to fixture_path

Thats fine we can implement `fixture_path`

> And I have to override setter method because of Rails 3 implementation of fixture_file_upload.

Also fine, including the file does that hence why delegation works at all..

To be clear I think we should remove the delegation.

Reply:
you're right, I'll do the change

Reply:
I'm not sure to get your point - `ActionController::TestCase` is a class, I don't see a really better way to inject it the `fixture_path`

Reply:
```Ruby
module RSpec
  module Rails
    # @private
    module FixtureFileUploadSupport
      extend ActiveSupport::Concern

      included do |other|
        if defined?(ActionDispatch::TestProcess)
          other.include ActionDispatch::TestProcess
        end
      end

      def fixture_path
        @fixture_path ||= calculate_path
      end

      def fixture_path=(value)
        # Rails 3.0..3.1 are using ActionController::TestCase class to resolve fixture_path
        # see https://apidock.com/rails/v3.0.0/ActionDispatch/TestProcess/fixture_file_upload
        if ActionController::TestCase.respond_to?(:fixture_path)
          ActionController::TestCase.fixture_path = value
        end
        @fixture_path = value
      end

      private

      def calculate_fixture_path
        resolved_fixture_path = (fixture_path || RSpec.configuration.fixture_path || '')
        self.fixture_path = File.join(resolved_fixture_path, '')
      end
    end
  end
end
```

Reply:
When `FixtureFileUploadSupport` is included it will define the relevant helpers.

Reply:
`fixture_path` has to be defined as a class method / attribute given how rails resolves it: `self.class.respond_to?(:fixture_path) && self.class.fixture_path`.
And directly including `ActionDispatch::TestProcess` will come with other methods not related like `cookies`, `session`, ... , I prefer to keep isolated from that class as much as possible.

Reply:
Oh that wasn't clear, in that case you need to do the definition inside another module and extend it:

```Ruby
module RSpec
  module Rails
    # @private
    module FixtureFileUploadSupport

      def self.included(other)
        if defined?(ActionDispatch::TestProcess)
          other.include ActionDispatch::TestProcess
        end
        other.extend FixturePathAttribute
      end

      module FixturePathAttribute

        def fixture_path
         @fixture_path ||= calculate_path
        end

        def fixture_path=(value)
          # Rails 3.0..3.1 are using ActionController::TestCase class to resolve fixture_path
          # see https://apidock.com/rails/v3.0.0/ActionDispatch/TestProcess/fixture_file_upload
          if ActionController::TestCase.respond_to?(:fixture_path)
            ActionController::TestCase.fixture_path = value
          end
          @fixture_path = value
        end

        private

        def calculate_fixture_path
          resolved_fixture_path = (fixture_path || RSpec.configuration.fixture_path || '')
          self.fixture_path = File.join(resolved_fixture_path, '')
        end
      end
    end
  end
end
```